website/content/en/blog/tracking-allocations.md

```diff
@@ -0,0 +1,50 @@
+
+## Trying it out
+
+![vector top with allocation tracing](/img/blog/vector-top-allocation-tracking.png)
+
+To explore the feature, start a Vector instance with the `allocation-tracing` option, launch `vector top` and monitor your components!
+
+## What's new
+
+We provide the following new metrics: `component_allocated_bytes`, `component_allocated_bytes_total`, and `component_deallocated_bytes_total`.
+
+- `component_allocated_bytes` captures the current net allocations/deallocations.
+- `component_allocated_bytes_total` shows the accumulated total allocations.
+- `component_deallocated_bytes_total` shows the accumulated total deallocations.
+
+The following image visualizes the `component_allocated_bytes` metric of a sample Vector instance:
+![Visualization of the allocation tracking metrics via Datadog](/img/blog/vector-allocation-tracking-graph.png)
+
+## How it works
+
+Under the hood, Vector uses a custom memory allocator implementation that captures each allocation/deallocation and associates it with the currently-executing component. This work builds upon Vector's existing tracing functionality. Allocations that aren't associated with any components are tracked by a `root` component.
+
+## Using allocation tracing in production
+
+Currently, this feature is only supported on Unix builds of Vector. This may change in the future as we continue to improve allocation tracing.
+
+Additionally, there is a performance overhead imposed by allocation tracing when enabled. Vector runs a set of soak tests in CI to catch performance regressions during development. These soak tests emulate common Vector use cases, such as remapping events, or converting payloads from one event type to another, and so on.
+
+In our development and testing of this feature, we've observed ~20% reduction in throughput. This overhead may be acceptable for debugging purposes, but care should be taken before enabling it across your entire fleet.
+
+## Next steps
+
+We currently do not provide support for determining shared memory usage between components. For example, when a sink is batching events, and is configured to batch many events, you may observe high Vector memory usage. If you looked at the memory usage metrics, you would see most of it being attributed to components that either created the events (such as a source) or processed the events (such as any transforms that modified the event) rather than the sink which is batching the events.
+
+In this example, Vector will attribute that memory usage to other components -- because that is where the initial allocations occurred -- but does not account for allocations moving to another component, where that new component is keeping the allocation alive (such as a sink batching events, as mentioned above).
+
+Adding support for shared ownership tracking provides further insights into the lifetimes of components, further easing the debugging process.
```

Comment:
This should be dropped, as the above text already describes the problem and that we plan to eventually tackle it.